app/models/station.rb

```diff
@@ -63,5 +63,62 @@ def self.most_rides_as_ending_place
     end.max.last[0].name
   end
 
- 
+ def self.number_of_starting_rides(station_id)
+  if Trip.group(:start_station_id).count[station_id]
+     Trip.group(:start_station_id).count[station_id]
+  else
+    return 0
+  end
+ end
+
+ def self.number_of_ending_rides(station_id)
+  if Trip.group(:end_station_id).count[station_id]
+   Trip.group(:end_station_id).count[station_id]
+  else
+    return 0
+  end
+ end
+
+ def self.most_frequent_destination_station(station_id)
+  if Station.find(station_id).trips_starting_here.group(:end_station_name).count.invert.max
+    Station.find(station_id).trips_starting_here.group(:end_station_name).count.invert.max.last
+  else
+    return "No trips from this station."
+  end
+ end
+
+ def self.most_frequent_origination_station(station_id)
+  if Station.find(station_id).trips_ending_here.group(:start_station_name).count.invert.max
+    Station.find(station_id).trips_ending_here.group(:start_station_name).count.invert.max.last
+  else
+    return "No trips to this station"
```

Comment:
@leepuppychow @timomitchel when I see strings in methods in the model, it is a code smell. That is view related information so you could put the if statement in the view. let me know if you have questions